sklearn/gaussian_process/gpr.py

```diff
@@ -0,0 +1,413 @@
+                # * 'initial_theta': the initial value for theta, which can be
+                #   used by local optimizers
+                # * 'bounds': the bounds on the values of theta
+                ....
+                # Returned are the best found hyperparameters theta and
+                # the corresponding value of the target function.
+                return theta_opt, func_min
+
+        Per default, the 'fmin_l_bfgs_b' algorithm from scipy.optimize
+        is used. If None is passed, the kernel's parameters are kept fixed.
+        Available internal optimizers are::
+
+            'fmin_l_bfgs_b'
+
+    n_restarts_optimizer: int, optional (default: 0)
+        The number of restarts of the optimizer for finding the kernel's
+        parameters which maximize the log-marginal likelihood. The first run
+        of the optimizer is performed from the kernel's initial parameters,
+        the remaining ones (if any) from thetas sampled log-uniform randomly
+        from the space of allowed theta-values. If greater than 0, all bounds
+        must be finite. Note that n_restarts_optimizer == 0 implies that one
+        run is performed.
+
+    normalize_y: boolean, optional (default: False)
+        Whether the target values y are normalized, i.e., the mean of the
+        observed target values become zero. This parameter should be set to
+        True if the target values' mean is expected to differ considerable from
+        zero. When enabled, the normalization effectively modifies the GP's
+        prior based on the data, which contradicts the likelihood principle;
+        normalization is thus disabled per default.
+
+    copy_X_train : bool, optional (default: False)
+        If True, a persistent copy of the training data is stored in the
+        object. Otherwise, just a reference to the training data is stored,
+        which might cause predictions to change if the data is modified
+        externally.
+
+    random_state : integer or numpy.RandomState, optional
+        The generator used to initialize the centers. If an integer is
+        given, it fixes the seed. Defaults to the global numpy random
+        number generator.
+
+    Attributes
+    ----------
+    X_train_ : array-like, shape = (n_samples, n_features)
+        Feature values in training data (also required for prediction)
+
+    y_train_: array-like, shape = (n_samples, [n_output_dims])
+        Target values in training data (also required for prediction)
+
+    kernel_: kernel object
+        The kernel used for prediction. The structure of the kernel is the
+        same as the one passed as parameter but with optimized hyperparameters
+
+    L_: array-like, shape = (n_samples, n_samples)
+        Lower-triangular Cholesky decomposition of the kernel in X_train_
+
+    alpha_: array-like, shape = (n_samples,)
+        Dual coefficients of training data points in kernel space
+    """
+    def __init__(self, kernel=None, sigma_squared_n=1e-10,
```

Comment:
fair enough, sounds like a sensible solution

On Tuesday, August 11, 2015, Jan Hendrik Metzen <notifications@github.com>
wrote:

> In sklearn/gaussian_process/gpr.py
> <https://github.com/scikit-learn/scikit-learn/pull/4270#discussion_r36782392>
> :
>
> > +        Feature values in training data (also required for prediction)
> > +
> > +    y_train_: array-like, shape = (n_samples, [n_output_dims])
> > +        Target values in training data (also required for prediction)
> > +
> > +    kernel_: kernel object
> > +        The kernel used for prediction. The structure of the kernel is the
> > +        same as the one passed as parameter but with optimized hyperparameters
> > +
> > +    L_: array-like, shape = (n_samples, n_samples)
> > +        Lower-triangular Cholesky decomposition of the kernel in X_train_
> > +
> > +    alpha_: array-like, shape = (n_samples,)
> > +        Dual coefficients of training data points in kernel space
> > +    """
> > +    def __init__(self, kernel=None, sigma_squared_n=1e-10,
>
> For GaussianProcessRegressor, what you mean used to be called
> sigma_squared_n and was never removed - I've renamed it now to alpha to
> be consistent with Ridge. jitter was the analogous thing for
> GaussianProcessClassifierand I removed it because it is very uncommon to
> set it to a non-zero value (because for classification, the observed values
> are not interpolated even for jitter=0 and jitter only affects the latent
> function modelled by the GP). For the rare cases where a jitter might be
> helpful, manually adding a WhiteKernel should be sufficient in my opinion.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/4270/files#r36782392>.
>